datafusion/sql/src/unparser/ast.rs

```diff
@@ -182,7 +182,20 @@ impl SelectBuilder {
         self
     }
     pub fn selection(&mut self, value: Option<ast::Expr>) -> &mut Self {
-        self.selection = value;
+        match (&self.selection, value) {
```

Comment:
This is unrelated but required to support table scan pushdowns. After we enabled push down filters to table scans we have plans where we have dedicated filter as previously and filters as part of table scand that processed separately and override main logical plan filter. This combines all filters to single `where` statement

Reply:
```sql
select
        c_phone as cntrycode,
        c_acctbal
from
        customer
where
        c_mktsegment = 'BUILDING'
        and c_acctbal > (
        select
                avg(c_acctbal)
        from
                customer);
```

```console
|               |   Federated                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
|               |  Projection: customer.c_phone AS cntrycode, customer.c_acctbal                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
|               |   Filter: CAST(customer.c_acctbal AS Decimal128(38, 6)) > (<subquery>)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
|               |     Subquery:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
|               |       Projection: avg(customer.c_acctbal)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |
|               |         Aggregate: groupBy=[[]], aggr=[[avg(customer.c_acctbal)]]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   |
|               |           TableScan: customer                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
|               |     TableScan: customer, full_filters=[customer.c_mktsegment = Utf8("BUILDING")] 
```

Reply:
I'll add unit test for this case and submit as a separate PR to datafusion main. Combining here to simplify testing/running benchmarks